src/freenet/node/NewPacketFormat.java

```diff
@@ -10,6 +10,7 @@
 import java.util.Iterator;
 import java.util.LinkedList;
 import java.util.List;
+import java.util.concurrent.ConcurrentSkipListMap;
```

Comment:
It's improving upon the HashMap... but yes, we could use a Deque or something else. Should we?

IMO the whole class should be re-factored... it doesn't make much sense as is... We could maintain reference to packets sorted by deadline instead.